Classes/YTPlayerView.m

```diff
@@ -299,23 +317,62 @@ - (void)setShuffle:(BOOL)shuffle {
 
 #pragma mark - Playback status
 
-- (float)videoLoadedFraction {
-  return [[self stringFromEvaluatingJavaScript:@"player.getVideoLoadedFraction();"] floatValue];
+- (void)videoLoadedFraction:(_Nullable YTFloatCompletionHandler)completionHandler {
+  [self stringFromEvaluatingJavaScript:@"player.getVideoLoadedFraction();"
+                     completionHandler:^(NSString *_Nullable result, NSError *_Nullable error) {
+    if (!completionHandler) {
+      return;
+    }
+
+    if (error) {
+      completionHandler(-1, error);
+      return;
+    }
+
+    completionHandler([result floatValue], error);
+  }];
 }
 
-- (YTPlayerState)playerState {
-  NSString *returnValue = [self stringFromEvaluatingJavaScript:@"player.getPlayerState();"];
-  return [YTPlayerView playerStateForString:returnValue];
+- (void)playerState:(_Nullable YTPlayerStateCompletionHandler)completionHandler {
+  [self stringFromEvaluatingJavaScript:@"player.getPlayerState();"
+                     completionHandler:^(NSString *_Nullable result, NSError *_Nullable error) {
+    if (!completionHandler) {
+      return;
+    }
+    if (error) {
+      completionHandler(kYTPlayerStateUnknown, error);
+      return;
+    }
+    completionHandler([YTPlayerView playerStateForString:result], nil);
+  }];
 }
 
-- (float)currentTime {
-  return [[self stringFromEvaluatingJavaScript:@"player.getCurrentTime();"] floatValue];
+- (void)currentTime:(_Nullable YTFloatCompletionHandler)completionHandler {
+  [self stringFromEvaluatingJavaScript:@"player.getCurrentTime();"
+                     completionHandler:^(NSString *_Nullable result, NSError *_Nullable error) {
+    if (!completionHandler) {
+      return;
+    }
+    if (error) {
+      completionHandler(-1, error);
+      return;
+    }
+    completionHandler([result floatValue], nil);
+  }];
 }
 
-// Playback quality
-- (YTPlaybackQuality)playbackQuality {
-  NSString *qualityValue = [self stringFromEvaluatingJavaScript:@"player.getPlaybackQuality();"];
-  return [YTPlayerView playbackQualityForString:qualityValue];
+- (void)playbackQuality:(_Nullable YTPlaybackQualityCompletionHandler)completionHandler {
```

Comment:
As I mentioned in the header file, all of the "playback quality" API calls have been removed from the iFrame player API. Please remove the the player-helper lib as well.

Reply:
done